Clan endpoints

iMobs opened the pull request.

Commit:
Stub out endpoints and add route to server

Commit:
Implement root route get/post

Commit:
Start to add tests and define expectations

Commit:
Add /clans/:clan endpoint and more tests

Commit:
Add more tests, more functionality for /clans

Updating Clan.find to take queries (or not).

mgarber93 commented on database/models/clan.js:
```diff
@@ -15,8 +15,8 @@ ClanModel.sync();
-Clan.findAll = function() {
-  return ClanModel.findAll();
+Clan.findAll = function(query = {}) {
+  return ClanModel.findAll({where: query});
```
Hmm interesting point, perhaps find should use findOne to limit results, and findAll can be used this way to have that functionality still exist. I can refactor Users, etc to follow this convention tomorrow.

Commit:
Correctly spell Barbara Streisand

Commit:
Fix tests after rebase

mgarber93 commented on server/clans.js:
```diff
@@ -0,0 +1,40 @@
+      res.status(201).json(newClan);
+    })
+    .catch(err => {
+      res.status(500).send(err.message);
```
We may want to send back a 400 here to indicate a malformed clan

Commit:
Refactor tests to be promise based

Commit:
Add members endpoint, move around gated data

Commit:
Some more endpoints fleshed out

These should probably be refactored (along with everything else) to not
allow volatile actions without authorization.

Commit:
Add update and delete endpoints

These are pretty rudimentary and don't really describe much.
